src/borg/testsuite/repository.py

```diff
@@ -1171,3 +1221,18 @@ def test_remote_borg_cmd(remote_repository):
         args.rsh = "ssh -i foo"
         remote_repository._args = args
         assert remote_repository.ssh_cmd(Location("ssh://example.com/foo")) == ["ssh", "-i", "foo", "example.com"]
+
+
+def test_remote_crash_before_compact():
+    # skip this test, we can't mock-patch a Repository class in another process!
+    pass
```

Comment:
I guess we could just as soon delete these last 3 tests? Their use as placeholders I think was a unittest thing to prevent execution of other tests with remote repo.

Reply:
The unittest-style test class for the remote repository tests had inherited most tests from the local repository test class, but some tests overwritten to not execute them because they can't work (see comments).

The comments might be worth keeping at the new place, where you do **not** have the remote_repo fixture now, to explain "why".

Reply:
Good point :)

Reply:
The comments in such "skipped" / empty tests still need to get moved to the right place.

The right place now is above or in the test function where you chose not to parametrize for both local and remote repos to get tested, but just tested for a local repo. So you can give a comment **there** to tell **why you didn't test with the remote repo**.

Reply:
That makes a lot of sense, thank you. I'll get that changed now.